File: Python/stats.py
from collections import Counter

class StatisticsCalculator:
    def __init__(self, data):
        self.data = sorted(data)

    def calculate_mean(self):
        return sum(self.data) / len(self.data)

    def calculate_median(self):
        n = len(self.data)
        if n % 2 == 0:
            return (self.data[n // 2 - 1] + self.data[n // 2]) / 2
        else:
            return self.data[n // 2]

    def calculate_mode(self):
        counts = Counter(self.data)
        max_count = max(counts.values())
        modes = [k for k, v in counts.items() if v == max_count]
        return modes if max_count > 1 else []

if __name__ == "__main__":
    data = list(map(int, input("Enter a list of integers separated by space: ").split()))
    stats = StatisticsCalculator(data)

    print(f"Mean: {stats.calculate_mean():.2f}")
    print(f"Median: {stats.calculate_median():.2f}")
    modes = stats.calculate_mode()
    if modes:
        print("Mode(s):", *modes)
    else:
        print("No mode found.")

File: C/stats.c
#include <stdio.h>
#include <stdlib.h>

#define MAX 100

// Function to compare two integers for qsort
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

float calculate_mean(int arr[], int n) {
    int sum = 0;
    for(int i = 0; i < n; i++)
        sum += arr[i];
    return (float)sum / n;
}

float calculate_median(int arr[], int n) {
    qsort(arr, n, sizeof(int), compare);
    if (n % 2 == 0)
        return (arr[n/2 - 1] + arr[n/2]) / 2.0;
    else
        return arr[n/2];
}

void calculate_mode(int arr[], int n) {
    int maxCount = 0;
    int modeValue[MAX];
    int modeIndex = 0;
    
    for(int i = 0; i < n; i++) {
        int count = 1;
        for(int j = i+1; j < n; j++) {
            if(arr[i] == arr[j])
                count++;
        }

        if(count > maxCount) {
            maxCount = count;
            modeValue[0] = arr[i];
            modeIndex = 1;
        } else if(count == maxCount && arr[i] != modeValue[modeIndex-1]) {
            modeValue[modeIndex++] = arr[i];
        }
    }

    if (maxCount == 1) {
        printf("No mode found.\n");
    } else {
        printf("Mode(s): ");
        for(int i = 0; i < modeIndex; i++)
            printf("%d ", modeValue[i]);
        printf("\n");
    }
}

int main() {
    int arr[MAX], n;

    printf("Enter the total count of numbers: ");
    scanf("%d", &n);

    printf("Enter %d integers:\n", n);
    for(int i = 0; i < n; i++)
        scanf("%d", &arr[i]);

    printf("Mean: %.2f\n", calculate_mean(arr, n));
    printf("Median: %.2f\n", calculate_median(arr, n));
    calculate_mode(arr, n);

    return 0;
}
